hw/ip/rom_ctrl/data/rom_ctrl.hjson

```diff
@@ -73,6 +73,27 @@
       desc: "A fatal error. Fatal alerts are non-recoverable and will be asserted until a hard reset."
     }
   ],
+
+  features: [
+    { name: "ROM_CTRL.SCRAMBLED",
+      desc: '''
+        ROM data is scrambled.
+      '''
+    },
+    { name: "ROM_CTRL.DIGESTS",
+      desc: '''
+        A signature of the ROM contents is computed at startup.
+        This is available in the DIGEST_<i> CSRs.
+      '''
+    },
+    { name: "ROM_CTRL.EXP_DIGESTS",
+      desc: '''
+        The chip has an expected ROM signature.
```

Comment:
Does the ROM control do anything with this expected signature? Or is it up to firmware to check that the digest and the expected digest match?

Reply:
Yep! It blocks boot if the two don't match. The idea is that an evil attacker would need to somehow subvert the machinery that stops the chip from starting at all, and would also find that the ROM signature gets incorporated into the creator root identity (not by firmware), so the chip wouldn't have a recognised identity anyway.